docs/infrastructure/logging.md

```diff
@@ -10,250 +10,272 @@ order: 6
+of the slick loggers to error level.
 
-The `cause` field may be omitted if the exception does not have another
-exception as its cause.
+Additionally, custom log events can be provided during runtime via system
+properties, without re-packaging the updated config file. For example
 
 ```typescript
-interface Exception {
-  // Name of the exception. In Java this can be the qualified classname.
-  name: String;
-  // Message associated with the exception. May be empty.
-  message: String;
-  // A stack trace indicating code location where the exception has originated
-  // from. May be empty if unavailable.
-  trace: [TraceElement];
-  // Optional, another exception that caused this one.
-  cause?: Exception;
-}
+akka.actor = info;
 ```
 
-##### `TraceElement`
-
-Represents a single element of exception's stacktrace.
+is equivalent to
 
 ```typescript
-interface TraceElement {
-  // Name of the stack location. For example, in Java this can be a qualified
-  // method name.
-  element: String;
-  // Code location of the element.
-  location: String;
-}
+  -Dakka.actor.Logger.level=info
 ```
 
-In Java, the location is usually a filename and line number locating the code
-that corresponds to the indicated stack location, for example `Main.java:123`.
-Native methods may be handled differently, as well as code from different
-languages, for example Enso also includes the columns - `Test.enso:4:3-19`.
+Any custom log level is therefore defined with `-Dx.y.Z.Logger.level` where `x`,
+`y` and `Z` refer to the package elements and class name, respectively. System
+properties always have a higher priority over those defined in the
+`application.conf` file.
 
-### Messages
+### Appenders
 
-Currently, the service supports only one message type - `LogMessage`, messages
-not conforming to this format will be ignored. The first non-conforming message
-for each connection will emit a warning.
+Log output target is also configured in the `application.conf` files in the
+"appenders" section ("appender" is equivalent to `java.util.logging.Handler`
+semantics). Each appender section can provide further required and optional
+key/value pairs, to better customize the log target output.
 
-#### `LogMessage`
+Currently supported are
 
-Describes the log message that the server should report and does not expect any
-response.
+- console appender - the most basic appender that prints log events to stdout
+- [file appender](#file-appender) - appender that writes log events to a file,
+  with optional rolling file policy
+- [socket appender](#socket-appender) - appender that forwards log events to
+  some logging server
+- [sentry.io appender](#sentry-appender) - appender that forwards log events to
+  a sentry.io service
 
-##### Parameters
+The appenders are defined by the `logging-service.appenders`. Currently only a
+single appender can be selected at a time. The selection may also be done via an
+environmental variable `$ENSO_DEFAULT_APPENDER`.
```

Comment:
Would it be possible to still have an stdout appender when selecting the file appender? While file appender is very useful, I think it's good to be able to still see the logs in the terminal.

We can do `tail -f logfile` but allowing the app to do this automatically would be ideal.

Reply:
I was going to add support for multiple appenders in the follow up PR